betterlockscreen

```diff
@@ -4,511 +4,859 @@
-	echo
-	echo '	-r --resolution'
-	echo '		to be used after -u'
-	echo '		used to set a custom resolution for the image cache.'
-	echo '		E.g: betterlockscreen -u path/to/image.png -r 1920x1080'
-	echo '		E.g: betterlockscreen -u path/to/image.png --resolution 3840x1080'
-	echo
-	echo
-	echo '	-b --blur'
-	echo '		to be used after -u'
-	echo '		used to set blur intensity. Default to 1.'
-	echo '		E.g: betterlockscreen -u path/to/image.png -b 3'
-	echo '		E.g: betterlockscreen -u path/to/image.png --blur 0.5'
-	echo
-	echo
-	echo '	-t --text'
-	echo '		to set custom lockscreen text (max 31 chars)'
-	echo "		E.g: betterlockscreen -l dim -t \"Don't touch my machine!\""
-	echo '		E.g: betterlockscreen --text "Hi, user!" -s blur'
-	echo
-	echo
-	echo '	--off <timeout>'
-	echo '		to set custom monitor turn off timeout for lockscreen'
-	echo '		timeout is in seconds'
-	echo '		E.g: betterlockscreen -l dim --off 5'
-	echo
-	echo
-	echo '	-wc --wallpaper_cmd <command>'
-	echo '		to set your custom wallpaper setter'
-	echo '		the default is "feh --bg-fill --no-fehbg"'
-	echo '		E.g: betterlockscreen -wc "xwallpaper --zoom" -w'
-	echo
-	echo
-	echo '	-tf --time_format <format>'
-	echo '		to set the time format used by i3lock-color'
-	echo '		see the i3lock or strftime man pages for more details.'
-	echo '		E.g: betterlockscreen -l dim -tf "%I:%M %p"'
+    echo
+    echo "Usage: betterlockscreen [-u <PATH>] [-l <EFFECT>] [-w <EFFECT>]"
+    echo
+    echo "  -u --update <PATH>"
+    echo "      Update lock screen image"
+    echo
+    echo "  -l --lock <EFFECT>"
+    echo "      Lock screen with cached image"
+    echo
+    echo "  -w --wall <EFFECT>"
+    echo "      Set wallpaper with cached image"
+    echo
+    echo "Additional arguments:"
+    echo
+    echo "  --display <N>"
+    echo "      Set display to draw loginbox"
+    echo
+    echo "  --span"
+    echo "      Scale image to span multiple displays"
+    echo
+    echo "  --off <N>"
+    echo "      Turn display off after N minutes"
+    echo
+    echo "  --fx <EFFECT,EFFECT,EFFECT>"
+    echo "      List of effects to generate"
+    echo
+    echo "  --desc <DESCRIPTION>"
+    echo "      Set a description for the new lock screen image"
+    echo "      (Only has an effect in combination with --update)"
+    echo
+    echo "  --show-layout"
+    echo "      Show current keyboard layout"
+    echo
+    echo "  --wallpaper-cmd <command>"
+    echo "      to set your custom wallpaper setter"
+    echo
+    echo "  --time-format <format>"
+    echo "      to set the time format used by i3lock-color"
+    echo
+    echo "  -- <ARGS>"
+    echo "      Pass additional arguments to i3lock"
+    echo
+    echo "Effects arguments:"
+    echo
+    echo "  --dim <N>"
+    echo "      Dim image N percent (0-100)"
+    echo
+    echo "  --blur <N>"
+    echo "      Blur image N amount (0.0-1.0)"
+    echo
+    echo "  --pixel <N,N>"
+    echo "      Pixelate image with N shrink and N grow (unsupported)"
```

Comment:
unsupported?

Reply:
@SebTM The default values are the only ones that produced a usable image for me, but I made it configurable anyways. Maybe someone will have better luck with it.